.circleci/real_config.yml

```diff
@@ -64,6 +64,11 @@ parameters:
     type: string
     default: ""
 
+det-versions: &det-versions
```

Comment:
@mackrorysd I need to figure out how we're going to sync this up to the list of supported releases. Can we talk about where this is going to live in the repo? Or how this list is bumped between releases?

Reply:
As I think about this, I think we'll need someone to do SOME manual step periodically no matter what. And that's fine because it's a small step. This person can be me - I have to maintain the release schedule, etc. anyway. If the step fits nicely into the release captain process it could be them. But either way, I think we need to do something manual when we "retire" a minor release line after 9 months or so.

So this list could be what we maintain and update directly, and we update it occasionally. I'm fine to own that. Or we could infer the list from the current list of branches and tags that fit a certain pattern (in which case, after 9 months we need to delete a branch that has hit EOL or rename it to NOT fit the pattern anymore).

What are your thoughts about those options?

Reply:
If we want to do this for a subset of releases, one option would be reading in a separate file that's a newline-delimited list of supported releases. At least then we wouldn't have to commit changes directly to the CircleCI config, and it kind of contains the blast radius a bit.

Another option would be to munge the output of `git tag` to filter out the three (or however many we want to support) latest minor releases, which shouldn't be too difficult.

A neat alternative would be to use commit notes, which can include arbitrary metadata, like `Is-Supported: true`. This requires annotated tags, though, and I wrote all of the new release stuff to use lightweight tags. I don't think that would be a difficult change, but it would be something for a bit further down the road.

Reply:
Regarding the `git tag` option, I could even update `version.sh` to provide a flag or two to return an arbitrary number of release versions back.

Reply:
+1 to David's idea to read a newline-delimited text from a separate file 
i don't expect release captains to have thorough knowledge of CircleCI, so it would be preferable to contain the changes OUTSIDE of this file.

regarding the `git tag` option, I think we'd have to connect @mackrorysd 's supported version logic to that and I don't think it's as simple as the last n-many minor releases. Please correct me if I'm wrong though.

Reply:
Yeah the supported versions will also vary between branches. Main wants to make sure we can upgrade from the latest commit on release branches, and patch releases want to make sure you can upgrade from their predecessors too.

I love the idea of a newline-delimited text file. Let's go with that!